HH-381 | Replace age range selector with single age selector

karisal-anders opened the pull request:
## Description

Replace age range selector with single age selector.

Also:
 - Fix the flaky subEvents test

### Related

- [HH-381](https://helsinkisolutionoffice.atlassian.net/browse/HH-381) Age selector
- [TH-1438](https://helsinkisolutionoffice.atlassian.net/browse/TH-1438) Fixing the flaky subEvents test

## Testing

### Automated tests

### Manual testing

## Screenshots

## Additional notes


[HH-381]: https://helsinkisolutionoffice.atlassian.net/browse/HH-381?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[TH-1438]: https://helsinkisolutionoffice.atlassian.net/browse/TH-1438?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

karisal-anders committed:
feat: add AdvancedSearchNumberInput, use it for selecting age in hobbies

Automatically limit age input to range 0–130. The upper limit is a
realistic value as 122 years is the oldest person documented to have
lived on earth according to Wikipedia:
https://en.wikipedia.org/wiki/List_of_the_verified_oldest_people

Also:
- Fix broken variable mapping in FilterSummary where maximum age was
  mapped to publisher, now publisher is mapped to publisher.

refs HH-381

karisal-anders committed:
fix(tests): fix flaky EventInfo subEvents test case

EventInfo's subEvents "should render sub events title and content when
ub events are given" test case was very flaky when running all
packages/components test locally. It failed almost always but not
always. When running just the EventInfo.test.tsx it passed all the time.

refs TH-1438 (ticket for fixing this particular flaky test)
refs HH-381 (needed to get tests working for this ticket)

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr847.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr847.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr847.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr847.dev.hel.ninja** :rocket::rocket::rocket:

karisal-anders commented on packages/components/src/components/advancedSearch/AdvancedSearchNumberInput.tsx:
```diff
@@ -0,0 +1,35 @@
+ * AdvancedSearchNumberInput is a component that provides a search input field
+ * with an icon—defaults to search icon if no other icon is specified.
+ * It is a pre-configured version of the AdvancedSearchInput
+ * component, using HDS TextInput and IconSearch.
```
```suggestion
 * component, using HDS NumberInput and IconSearch.
```

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr847.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-hobbies-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr847.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr847.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-events-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**TAPAHTUMAT-UI branch is deployed to platta: https://tapahtumat-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**HARRASTUKSET-UI branch is deployed to platta: https://harrastukset-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr847.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr847.dev.hel.ninja** :rocket::rocket::rocket:

karisal-anders commented:
TODO:
 - [ ] Refactoring AgeSelector props

Requesting code review already.

karisal-anders committed:
feat(components): add optional icon parameter to AdvancedSearchTextInput

Now the icon can be overridden in AdvancedSearchTextInput component
similarly as in AdvancedSearchNumberInput.

refs HH-381

karisal-anders commented on apps/hobbies-helsinki/src/domain/search/eventSearch/ageSelector/AgeSelector.tsx:
```diff
@@ -0,0 +1,36 @@
+      value={ageInput}
+    />
+  );
+};
```
Removed the refactoring commit that moved this to AgeSelector.tsx. Now it is used directly in AdvancedSearchForm instead.

terovirtanen commented:
**FEDERATION-ROUTER branch is deployed to platta: https://events-graphql-federation-sports-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr847.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**EVENTS-GRAPHQL-PROXY branch is deployed to platta: https://events-graphql-proxy-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**VENUE-GRAPHQL-PROXY branch is deployed to platta: https://venue-graphql-proxy-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**LIIKUNTA-UI branch is deployed to platta: https://liikunta-pr847.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr847.dev.hel.ninja :laughing::tada::tada::tada:
